src/transport/tcp/connection.rs

```diff
@@ -140,6 +137,16 @@ impl NegotiatedConnection {
     }
 }
 
+/// Connection specific metrics.
+pub struct TcpConnectionMetrics {
+    /// Metric for the number of pending substreams that are negotiated.
+    pub pending_substreams_num: MetricGauge,
+
+    /// Metric incremented when the connection starts
+    /// and decremented when the connection closes.
+    pub _active_connections_num: ScopeGaugeMetric,
```

Comment:
nit: starting with an underscore looks confusing, like it's not implemented or so. May be `#[allow_unused]`?